src/GoogleFit/FitOnFhir.GoogleFit/Client/Handlers/GoogleFitAuthorizationHandler.cs

```diff
@@ -112,5 +117,31 @@ private async Task<IActionResult> Callback(RoutingRequest request)
                 return new NotFoundObjectResult(ex.Message);
             }
         }
+
+        private async Task<IActionResult> Revoke(RoutingRequest request)
+        {
+            string patientId;
+            string system;
+            try
+            {
+                patientId = HttpUtility.UrlDecode(EnsureArg.IsNotNullOrWhiteSpace(request?.HttpRequest?.Query?[Constants.PatientIdQueryParameter], $"query.{Constants.PatientIdQueryParameter}"));
+                system = HttpUtility.UrlDecode(EnsureArg.IsNotNullOrWhiteSpace(request?.HttpRequest?.Query?[Constants.SystemQueryParameter], $"query.{Constants.SystemQueryParameter}"));
+            }
+            catch (ArgumentException)
+            {
+                return new BadRequestObjectResult($"'{Constants.PatientIdQueryParameter}' and '{Constants.SystemQueryParameter}' are required query parameters.");
+            }
+
+            try
+            {
+                return await _usersService.RevokeAccess(patientId, system, request.Token) ?
+                    new OkObjectResult("Access revoked successfully.") : new NotFoundObjectResult("Access not revoked");
```

Comment:
See my other comment about not using a bool return type in conjunction with this.

I think the failure status code is a 500 because in theory everything should work, so if they encounter this issue there was an internal error somewhere along the the line. Also, I think the message should be a bit more "production quality". Maybe "An unexpected error occurred while attempting to revoke data access."